lib/src/verifiers/utilities/json-schema-verifier.spec.ts

```diff
@@ -0,0 +1,603 @@
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+
+      it("validates literal", () => {
+        const dataType: StringLiteral = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates literal", () => {
+        const dataType: StringLiteral = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "not hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+    });
+
+    describe("number schema", () => {
+      it("validates", () => {
+        const dataType: NumberType = {
+          kind: TypeKind.NUMBER
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates", () => {
+        const dataType: NumberType = {
+          kind: TypeKind.NUMBER
+        };
+        const data: StringExpression = {
+          kind: TypeKind.STRING_LITERAL,
+          value: "hello"
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+
+      it("validates literal", () => {
+        const dataType: NumberLiteral = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates literal", () => {
+        const dataType: NumberLiteral = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 54
+        };
+        const data: NumberExpression = {
+          kind: TypeKind.NUMBER_LITERAL,
+          value: 53
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).toThrow();
+      });
+    });
+
+    describe("array schema", () => {
+      it("validates", () => {
+        const dataType: ArrayType = {
+          kind: TypeKind.ARRAY,
+          elements: {
+            kind: TypeKind.STRING
+          }
+        };
+        const data: ArrayExpression = {
+          kind: TypeKind.ARRAY,
+          elements: [
+            {
+              kind: TypeKind.STRING_LITERAL,
+              value: "hello"
+            },
+            {
+              kind: TypeKind.STRING_LITERAL,
+              value: "bye"
+            }
+          ]
+        };
+        expect(() => verifyJsonSchema(dataType, data, [])).not.toThrow();
+      });
+
+      it("invalidates", () => {
```

Comment:
I think it would be worth having slightly more specific names for each test case, e.g. `invalidates incorrect item type in list` or `validates empty list`.